jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/NodeInstanceImpl.java

```diff
@@ -393,7 +393,7 @@ public void triggerNode(long nodeId) {
     	org.jbpm.workflow.instance.NodeInstance nodeInstance = (org.jbpm.workflow.instance.NodeInstance)
     		((org.jbpm.workflow.instance.NodeInstanceContainer) getNodeInstanceContainer())
             	.getNodeInstance(getNode().getNodeContainer().getNode(nodeId));
-    	triggerNodeInstance(nodeInstance, null);
+    	triggerNodeInstance(nodeInstance, org.jbpm.workflow.core.Node.CONNECTION_DEFAULT_TYPE);
```

Comment:
@krisv the reason is that some of the NodeInstances (e.g. SubProcessNodeInstance or TimerNodeInstance) throw exception if the connection is different than the default. So instead of fixing all the node instance to check if it's null I thought it's better to simply provide the default connection when retriggering.